sike-java/src/main/java/com/wultra/security/pqc/sike/model/EncryptedMessage.java

```diff
@@ -73,7 +73,7 @@ public EncryptedMessage(SikeParam sikeParam, byte[] bytes) {
      */
     public byte[] getEncoded() {
         if (c0 == null || c1 == null) {
-            return null;
+            return new byte[0];
```

Comment:
@romanstrobl Please review the semantics here, maybe we throw an error somewhere in case of null?

Reply:
Although this change does not impact current code, I do not like the change, because a comparison function which compares two messages encoded using `getEncoded()` could consider them as equal if both where `new byte[0]`. I prefer to stick with the `null` value for this reason.

Reply:
I was paying attention to [MSC19-C](https://wiki.sei.cmu.edu/confluence/display/c/MSC19-C.+For+functions+that+return+an+array%2C+prefer+returning+an+empty+array+over+a+null+value) when I made the change. In this codebase, returning null is not an issue, but in the future it will force callers of getEncoded() to test for nullity, making code less readable and maybe more complex (that was at least my thought).

Reply:
Changed on 7b01fa3fd7ae38575e3917d7b4addad39f255ed3